src/Microsoft.Health.Dicom.Core/Features/CustomTag/GetAllCustomTagsHandler.cs

```diff
@@ -0,0 +1,31 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Dicom.Core.Messages.CustomTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.CustomTag
+{
+    public class GetAllCustomTagsHandler : IRequestHandler<GetAllCustomTagsRequest, GetAllCustomTagsResponse>
+    {
+        private readonly IGetCustomTagsService _getCustomTagsService;
+
+        public GetAllCustomTagsHandler(IGetCustomTagsService getCustomTagsService)
+        {
+            EnsureArg.IsNotNull(getCustomTagsService, nameof(getCustomTagsService));
+            _getCustomTagsService = getCustomTagsService;
+        }
+
+        public async Task<GetAllCustomTagsResponse> Handle(GetAllCustomTagsRequest request, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+
+            return await _getCustomTagsService.GetAllCustomTagsAsync(cancellationToken);
```

Comment:
>return await [](start = 12, length = 12)

nit: This method does not need to be `async`. Instead you can simply say `return _getCustomTagsService.GetAllCustomTagsAsync(cancellationToken);` to return the `Task<T>` directly

Reply:
It looks like our existing handlers and services always follow this pattern. Do you think we will increase latencies by doing this?

---
In reply to: [564813493](https://github.com/microsoft/dicom-server/pull/501#discussion_r564813493) [](ancestors = 564813493)

Reply:
No, it is just unnecessary. The IL is much more complex in this case, as the state machine needs to be constructed and the task must be fetched (essentially you must create a Task around the Task). You can leave this as-is, but we probably want to avoid this in the future just to avoid the overhead.

---
In reply to: [566992864](https://github.com/microsoft/dicom-server/pull/501#discussion_r566992864) [](ancestors = 566992864,564813493)